src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UseConcreteTypeTests.cs

```diff
@@ -0,0 +1,614 @@
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.UseConcreteTypeAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public static partial class UseConcreteTypeTests
+    {
+        [Fact]
+        public static async Task Params()
+        {
+            const string Source = @"
+                namespace Example
+                {
+                    public interface IFoo
+                    {
+                        void Bar();
+                    }
+
+                    public class Foo : IFoo
+                    {
+                        public void Bar() {}
+                    }
+
+                    public class Test
+                    {
+                        private void Method(IFoo {|#0:foo|})
+                        {
+                            foo.Bar();
+                        }
+
+                        private void Method2(IFoo foo)
+                        {
+                            foo.Bar();
+                        }
+
+                        private void Caller(IFoo ifoo)
+                        {
+                            Method(new Foo());
+                            Method2(new Foo());
+                            Method2(ifoo);
+                        }
+                    }
+                }";
+
+            await TestCSAsync(Source,
+                VerifyCS.Diagnostic(UseConcreteTypeAnalyzer.UseConcreteTypeForParameter)
+                        .WithLocation(0)
+                        .WithArguments("foo", "Example.IFoo", "Example.Foo"));
+        }
+
+        [Fact]
+        public static async Task Conditional()
+        {
+            const string Source = @"
+            #nullable enable
+            namespace Example
+            {
+                public interface IFoo
+                {
+                    void Bar();
+                }
+
+                public class Foo : IFoo
+                {
+                    public void Bar() {}
+                }
+
+                public class FooProvider
+                {
+                    public Foo Foo { get { return new Foo(); } }
+                }
+
+                public class AttributeData
+                {
+                    public SyntaxReference? ApplicationReference { get; }
+                }
+
+                public abstract class SyntaxReference
+                {
+                    public abstract SyntaxNode GetSyntax();
+                }
+
+                public abstract class SyntaxNode
+                {
+                    public Location GetLocation() => new();
+                }
+
+                public class Location
+                {
+                }
+
+                public class Test
+                {
+                    private IFoo? {|#0:Method1|}(FooProvider? provider)
```

Comment:
Same here and others, `private {|#0:IFoo|}? Method1(FooProvider? provider)` not `private IFoo? {|#0:Method1|}(FooProvider? provider)`